pkg/cloudprovider/provider/anexia/network_provisioning.go

```diff
@@ -0,0 +1,122 @@
+package anexia
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/kubermatic/machine-controller/pkg/apis/cluster/common"
+	anxtypes "github.com/kubermatic/machine-controller/pkg/cloudprovider/provider/anexia/types"
+	anxclient "go.anx.io/go-anxcloud/pkg/client"
+	anxaddr "go.anx.io/go-anxcloud/pkg/ipam/address"
+	anxvm "go.anx.io/go-anxcloud/pkg/vsphere/provisioning/vm"
+	"go.uber.org/zap"
+)
+
+func networkInterfacesForProvisioning(ctx context.Context, log *zap.SugaredLogger, client anxclient.Client) ([]anxvm.Network, error) {
+	reconcileContext := getReconcileContext(ctx)
+
+	config := reconcileContext.Config
+	status := reconcileContext.Status
+
+	if status.Networks == nil || len(status.Networks) != len(config.Networks) {
+		if len(status.Networks) != 0 && len(status.Networks) != len(config.Networks) {
+			log.Warn("size of status.Networks != config.Networks, this should not happen in normal operation - ignoring existing status")
+		}
+
+		status.Networks = make([]anxtypes.NetworkStatus, len(config.Networks))
+	}
```

Comment:
It's pretty hard to keep the context in the brain when reading those lines. Can we simplify the branches a bit, maybe by using a `switch` statement? I am thinking about sth along these lines:

```suggestion
	switch {
	// Case 1: we already have networks, but it differs from the config.
	case len(status.Networks) != 0 && len(status.Networks) != len(config.Networks):
		log.Warn("size of status.Networks != config.Networks, this should not happen in normal operation - ignoring existing status")
	// Case 2: There are networks configured and we have to setup them.
	case len(config.Networks) > 0:
		status.Networks = make([]anxtypes.NetworkStatus, len(config.Networks))
	// Case 3: Do nothing otherwise.
	default:
	}
```

Reply:
that's a different behavior though, irrelevant difference, but still

I also don't think this is more readable .. I will add a nice comment though, maybe that helps already?

Reply:
actually found something to make better with those if statements - see force-pushed changes in a bit :)